Automattic/wp-calypso
# BackupRetention: Add tracks events for selection options and submit

Resolves #73354 

## Proposed Changes

* Add event `calypso_jetpack_backup_storage_retention_option_click` to track when user clicks on a retention option.
* Add event `calypso_jetpack_backup_storage_retention_submit_click` to track when user clicks on the submit button.

## Testing Instructions
* Start a Jetpack Cloud live branch
* Select a site with a backup plan that has storage limit like Jetpack Backup 10 GB. 
* Navigate to Settings and wait for the new setting to render. It should render.
* Open developer console in Network tab and ensure `Preserve log` are ticked.
* Click on any retention option radio button.
* You should see a `t.gif` call with the event `calypso_jetpack_backup_storage_retention_option_click` including the prop `retention_option` with the selected number of days.
* Click on any retention option radio button that doesn't require a storage upgrade.
* Click on `Update settings` button.
* You should see a `t.gif` call with the event `calypso_jetpack_backup_storage_retention_submit_click` including the prop `retention_option` with the selected number of days.

## Pre-merge Checklist
